### PR TITLE
Email autocomplete from dev

### PR DESCRIPTION
**Pull request by NorbertPointu**
Add autocompelete

Issue : 
When the autocomplete is displaying we see the back screen :
image

When we select an item, the input selection is not cleared

### CODE REVIEW

**Review by avine — CHANGES_REQUESTED** [review]

File: server/src/feedback/feedback.dto.ts [hunk 1]
```diff
@@ -1,37 +1,72 @@
...
-  @IsEmail() @Transform((params) => (params.value as string).toLowerCase()) recipient: string;
+  @ApiProperty({ description: 'Recipient email' })
+  @IsEmail()
+  @Transform((params) => (params.value as string).toLowerCase())
+  recipient: string;
 
-  @IsString() @MaxLength(MEDIUM_MAX_LENGTH) message: string;
+  @IsString()
+  @MaxLength(MEDIUM_MAX_LENGTH)
+  @ApiProperty({ description: 'message' })
```
[the remark this answers is not in the document]
Comment (NorbertPointu) [reply server/src/feedback/feedback.dto.ts:14]: Move to be the first decorator in all places

File: server/src/employee/employee.controller.ts [hunk 2]
```diff
@@ -1,9 +1,12 @@
-import { BadRequestException, Body, Controller, Get, Post, UseGuards } from '@nestjs/common';
+import { BadRequestException, Body, Controller, Get, Post, Query, UseGuards } from '@nestjs/common';
+import { ApiBearerAuth } from '@nestjs/swagger';
+import { google } from 'googleapis';
 import { AuthGuard, AuthService } from '../core/auth';
 import { EmployeeDbService } from './employee-db';
 import { UpdateManagerDto } from './employee.dto';
 import { buildRequiredEmployeeData } from './employee.utils';
 
+@ApiBearerAuth('defaultBearerAuth')
```
[the remark this answers is not in the document]
Comment (NorbertPointu) [reply server/src/employee/employee.controller.ts]: le nom "defaultBearerAuth"  a été supprimé

File: server/src/employee/employee.controller.ts [hunk 3]
```diff
@@ -27,4 +30,10 @@ export class EmployeeController {
     }
     return this.employeeDbService.updateManager(employeeEmail, managerEmail);
   }
+
+  @Get('search')
+  @UseGuards(AuthGuard)
+  async search(@Query('input') searchInput: string) {
+    return this.employeeDbService.searchEmployee(searchInput);
```
[the remark this answers is not in the document]
Comment (NorbertPointu) [reply server/src/employee/employee.controller.ts:36]: already changed

File: server/src/employee/employee-db/employee-db.service.ts [hunk 4]
```diff
@@ -61,4 +62,111 @@ export class EmployeeDbService {
...
+          },
+        ],
+        totalSize: 2,
+      },
+    };
+
+    const resultRawData = mockData;
+
+    return resultRawData.data.people!.map<EmployeeSearchResult>(({ names, photos, emailAddresses }) => ({
+      displayName: names?.[0].displayName ?? 'Not defined',
```
[the remark this answers is not in the document]
Comment (NorbertPointu) [reply server/src/employee/employee-db/employee-db.service.ts:168]: will be changed when the api will be on

File: server/src/core/auth/auth.service.ts [hunk 5]
```diff
@@ -18,6 +18,7 @@ export class AuthService {
 
   async authenticateUser(idToken?: string): Promise<void> {
     try {
+
```
[the remark this answers is not in the document]
Comment (NorbertPointu) [reply server/src/core/auth/auth.service.ts]: done, add format on save in my vscode

**Pull request closed without merging**